source/connection-monitoring-and-pooling/tests/cmap-format/pool-checkout-error-network.yml

```diff
@@ -0,0 +1,39 @@
+version: 1
+style: integration
+description: must properly handle network error on check out
+runOn:
+  -
+    # required for appName in fail point
+    minServerVersion: "4.9.0"
+failPoint:
+  configureFailPoint: failCommand
+  # high amount to ensure not interfered with by monitor checks.
+  mode: { times: 50 }
+  data:
+    failCommands: ["isMaster","hello"]
+    closeConnection: true
+    appName: &appName "poolCheckOutErrorTest"
+poolOptions:
+  appName: *appName
+operations:
+  - name: ready
+  - name: checkOut
+error:
+  type: NetworkError
+events:
+  - type: ConnectionCheckOutStarted
+    address: 42
+  - type: ConnectionCreated
+    address: 42
+  - type: ConnectionPoolCleared
+    address: 42
+  - type: ConnectionClosed
+    address: 42
+    connectionId: 1
+    reason: error
+  - type: ConnectionCheckOutFailed
+    address: 42
+    reason: connectionError
+ignore:
+  - ConnectionPoolCreated
+  - ConnectionPoolReady
```

Comment:
This test fails in Python. For example the python cmap-format test runner mocks various pieces of the MongoClient and Topology to get a connection Pool suitable for the tests, but this test expects to observe the effects of the Topology's error handling rules. I think this kind of test is outside the scope of the cmap test format and has to be implemented in the unified runner. Any test for the interaction between SDAM and CMAP needs to be moved over to the unified runner IMO. Does that make sense?

Reply:
@BorisDog, how does this test pass in .NET?

Reply:
This is an [integration tests format](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/tests/cmap-format/README.rst#integration-test-format), so it runs against real deployment, with real SDAM in .NET.

Reply:
I'm not sure using SDAM here is expected. If SDAM is running then the heartbeats will fail and clear the pool at any time, thus the events would be non-deterministic.

Reply:
Yeah we hack it a bit: we set 10 mins heartbeat interval, and then force the pool to mark itself as ready.

Reply:
I suppose I can implement more hacks to get this to "work" but at that point it wouldn't even test the same code path that the driver takes for handling an error while checking out a connection for an operation. Is there any reason _not_ to move these tests over to the unified format?

Reply:
In C# it's just the initial setup that is adjusted, so the rest of SDAM works as usual.

But I agree that if unified format fits better, than it's preferable.
BTW this PR already contains an unified test similar to this one: ["Pool properly handles network error during checkout"](https://github.com/mongodb/specifications/pull/1369/files/bec7969f068b7e1778bd6929d4ae9a7157e2b749#diff-f83bf8d2e7fc7772a5452ed362a7ddc1d4a888036ee8e527bd49b92a1a322104R100), not sure if they test the exact same thing...

Reply:
Good point. Yes it looks like that test is exactly what I'm asking for. Let's remove this test. Let me check which other cmap-format tests are problematic.

Reply:
Sorry for letting this one simmer for a while--I've gone ahead and updated all the newly introduced CMAP-format tests to the unified format, with the exception of "checking in a connection after checkout fails closes connection" as mentioned above.